ee/tabby-webserver/src/service/license.rs

```diff
@@ -48,6 +49,51 @@ pub fn validate_license(token: &str) -> Result<LicenseInfo, jwt::errors::ErrorKi
     Ok(data?.claims)
 }
 
+fn jwt_timestamp_to_utc(f: f64) -> Result<DateTime<Utc>> {
+    let secs = f as i64;
+    Ok(NaiveDateTime::from_timestamp_millis(secs * 1000)
+        .ok_or_else(|| anyhow!("Timestamp is corrupt"))?
+        .and_utc())
+}
+
+#[async_trait]
+impl LicenseService for DbConn {
+    async fn get_license_info(&self) -> Result<Option<LicenseInfo>> {
+        let Some(license) = self.read_enterprise_license().await? else {
+            return Ok(None);
+        };
+        let license =
+            validate_license(&license).map_err(|e| anyhow!("License is corrupt: {e:?}"))?;
+
+        let issued_at = jwt_timestamp_to_utc(license.iat)?;
+        let expires_at = jwt_timestamp_to_utc(license.exp)?;
+
+        let status = if expires_at < Utc::now() {
+            LicenseStatus::Expired
+        } else {
+            LicenseStatus::Ok
+        };
+
+        let license = LicenseInfo {
+            r#type: LicenseType::TEAM,
+            status,
+            seats: license.num as i32,
+            issued_at,
+            expires_at,
+        };
+
+        Ok(Some(license))
+    }
+
+    async fn update_license(&self, license: Option<String>) -> Result<()> {
+        if let Some(license) = &license {
+            validate_license(license).map_err(|e| anyhow!("License is corrupt: {e:?}"))?;
```

Comment:
reuse logic inside of `get_license_info`, to acquire the `LicenseStatus`, thus generating the proper error message for different status (e.g expired, exceed seat, ...)